alf/algorithms/ddpg_algorithm.py

```diff
@@ -252,8 +253,8 @@ def _critic_train_step(self, exp: Experience, state: DdpgCriticState):
         target_q_values, target_critic_states = self._target_critic_networks(
             (exp.observation, target_action), state=state.target_critics)
 
-        if target_q_values.ndim == 3 and self._reward_weights is not None:
-            sign = self._reward_weights.sign()
+        if self.has_multidim_reward():
+            sign = self.reward_weights.sign()
```

Comment:
Is it ``self._reward_weights ``?

Reply:
There is a property defined in RLAlgorithm base class.